editor/reducer.js

```diff
@@ -236,6 +236,38 @@ export const editor = flow( [
 				];
 			}
 
+			case 'MOVE_BLOCK_TO_INDEX': {
+				if ( ! Number.isInteger( action.index ) ) {
+					return state;
+				}
+
+				if ( ! state.length || action.index > state.length - 1 || action.index < 0 ) {
+					return state;
+				}
+
+				const blockIndex = state.indexOf( action.uid );
+
+				if ( blockIndex === -1 || blockIndex === action.index ) {
+					return state;
+				}
+
+				return without( state.reduce( ( acc, elem, i ) => {
```

Comment:
Is the `without` necessary here? Couldn't we just not return with the `null` in the `Array#reduce` accumulator if we don't want it to be included?

```js
if ( blockIndex === i ) {
	return acc;
}
```